components/url_sanitizer/browser/url_sanitizer_service.cc

```diff
@@ -0,0 +1,155 @@
+    const std::string& json) {
+  base::JSONReader::ValueWithError parsed_json =
+      base::JSONReader::ReadAndReturnValueWithError(json);
+  if (!parsed_json.value || !parsed_json.value->is_list()) {
+    VLOG(1) << "Error parsing feature JSON: " << parsed_json.error_message;
+    return {};
+  }
+  base::flat_set<std::unique_ptr<URLSanitizerService::MatchItem>> matchers;
+  for (const auto& it : parsed_json.value->GetList()) {
+    auto* include_list = it.FindListPath("include");
+    if (!include_list)
+      continue;
+    extensions::URLPatternSet include_matcher;
+    if (!CreateURLPatternSetFromValue(include_list, &include_matcher))
+      continue;
+    auto* params_list = it.FindListPath("params");
+    base::flat_set<std::string> params;
+    if (!CreateParamsList(params_list, &params))
+      continue;
+
+    extensions::URLPatternSet exclude_matcher;
+    CreateURLPatternSetFromValue(it.FindListPath("exclude"), &exclude_matcher);
+    auto item = std::make_unique<URLSanitizerService::MatchItem>(
+        std::move(include_matcher), std::move(exclude_matcher), params);
+
+    matchers.insert(std::move(item));
+  }
+
+  return matchers;
+}
+
+}  // namespace
+
+URLSanitizerService::URLSanitizerService() = default;
+
+URLSanitizerService::~URLSanitizerService() = default;
+
+URLSanitizerService::MatchItem::MatchItem() = default;
+URLSanitizerService::MatchItem::~MatchItem() = default;
+
+URLSanitizerService::MatchItem::MatchItem(extensions::URLPatternSet in,
+                                          extensions::URLPatternSet ex,
+                                          base::flat_set<std::string> prm) {
+  include = std::move(in);
+  exclude = std::move(ex);
+  params = std::move(prm);
+}
+
+void URLSanitizerService::Initialize(const std::string& json) {
+  base::ThreadPool::PostTaskAndReplyWithResult(
+      FROM_HERE, {base::MayBlock()}, base::BindOnce(&ParseFromJson, json),
+      base::BindOnce(&URLSanitizerService::UpdateMatchers,
+                     weak_factory_.GetWeakPtr()));
+}
+
+void URLSanitizerService::UpdateMatchers(
+    base::flat_set<std::unique_ptr<URLSanitizerService::MatchItem>> mappings) {
+  matchers_ = std::move(mappings);
+  if (initialization_callback_for_testing_)
+    std::move(initialization_callback_for_testing_).Run();
+}
+
+GURL URLSanitizerService::SanitizeURL(const GURL& initial_url) {
+  if (matchers_.empty())
+    return GURL();
+  GURL url = initial_url;
+  for (const auto& it : matchers_) {
+    if (!it->include.MatchesURL(url) || it->exclude.MatchesURL(url))
+      continue;
+    std::string debug;
+    for (const auto& p : it->params) {
+      debug += "," + std::string(p);
+    }
+    auto sanitized_query = StripQueryParameter(url.query(), it->params);
+    GURL::Replacements replacements;
+    replacements.SetQueryStr(sanitized_query);
+    url = url.ReplaceComponents(replacements);
+  }
+  return url;
+}
+
+std::string URLSanitizerService::StripQueryParameter(
```

Comment:
I'd suggest a comment along the lines of:
```
// FIXME: merge with browser/net/brave_site_hacks_network_delegate_helper.cc::StripQueryParameter()
```
just so we don't lose track of this future cleanup.

Reply:
done